### PR TITLE
[PM-3996] Scaffolding for preload script

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

This PR wires up a polyfill for `window.ipc` which allows us to progressively migrate the codebase to a format which supports context bridge. This avoids a big bang effort where every non sandboxed call has to be migrated before we can run the code.

Once all calls to node modules are removed from the renderer and only exists in preload.ts. We will turn on context isolation and use the context bridge for communication instead.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Review by dani-garcia — DISMISSED** [review]

File: apps/desktop/src/app/main.ts [hunk 1]
```diff
@@ -1,8 +1,12 @@
 import { enableProdMode } from "@angular/core";
 import { platformBrowserDynamic } from "@angular/platform-browser-dynamic";
 
+import { ipc } from "../preload";
 import { isDev } from "../utils";
 
+// Temporary polyfill for preload script
```
Comment (MGibson1) [inline apps/desktop/src/app/main.ts:7]: What's the plan to remove? why's it temporary?

File: apps/desktop/src/main/window.main.ts [hunk 2]
```diff
@@ -143,6 +143,7 @@ export class WindowMain {
       backgroundColor: await this.getBackgroundColor(),
       alwaysOnTop: this.enableAlwaysOnTop,
       webPreferences: {
+        // preload: path.join(__dirname, "preload.js"),
```
Comment (MGibson1) [inline apps/desktop/src/main/window.main.ts:146]: Doesn't this need to be uncommented?

File: apps/desktop/src/preload.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ * https://www.electronjs.org/docs/latest/tutorial/tutorial-preload
+ * https://www.electronjs.org/docs/latest/api/context-bridge
+ */
+
+// Each team owns a subspace of the `ipc` global variable in the renderer.
+export const ipc = {
+  platform,
+};
+
+// contextBridge.exposeInMainWorld("ipc", ipc);
```
Comment (MGibson1) [inline apps/desktop/src/preload.ts:19]: What's this?

**Review by MGibson1 — DISMISSED** [review]

File: apps/desktop/src/app/main.ts (hunk 1 above)
Comment (Hinton) [reply apps/desktop/src/app/main.ts:7]: I updated the PR description to include this information.

**Review by dani-garcia — APPROVED** [review]

**Review by djsmith85 — APPROVED** [review]
Approving the merge from master, as @MGibson1 had previously approved for @bitwarden/tech-leads.

**Pull request merged**